frontend/web/components/modals/CreateMetadata.tsx

```diff
@@ -0,0 +1,458 @@
+            <div className='entities-cell'>
+              <Switch
+                checked={environmentEnabled}
+                onChange={() => {
+                  setEnvironmentEnabled(!environmentEnabled)
+                  handleMetadataModelField(
+                    environmentContentType,
+                    !environmentEnabled,
+                  )
+                }}
+                className='ml-0'
+              />
+            </div>
+            <div className='entities-cell'>
+              <Switch
+                checked={environmentRequired}
+                onChange={() => {
+                  setEnvironmentRequired(!environmentRequired)
+                  handleRequiredMetadataModelField(
+                    environmentContentType,
+                    !environmentRequired,
+                  )
+                }}
+                className='ml-0'
+              />
+            </div>
+          </div>
+          <div className='entities-row'>
+            <div className='entities-cell entities-header'>Segment</div>
+            <div className='entities-cell'>
+              <Switch
+                checked={segmentEnabled}
+                onChange={() => {
+                  setSegmentsEnabled(!segmentEnabled)
+                  handleMetadataModelField(segmentContentType, !segmentEnabled)
+                }}
+                className='ml-0'
+              />
+            </div>
+            <div className='entities-cell'>
+              <Switch
+                checked={segmentRequired}
+                onChange={() => {
+                  setSegmentRequired(!segmentRequired)
+                  handleRequiredMetadataModelField(
+                    segmentContentType,
+                    !segmentRequired,
+                  )
+                }}
+                className='ml-0'
+              />
+            </div>
+          </div>
+          <div className='entities-row'>
+            <div className='entities-cell entities-header'>Flag</div>
+            <div className='entities-cell'>
+              <Switch
+                checked={flagEnabled}
+                onChange={() => {
+                  setFlagsEnabled(!flagEnabled)
+                  handleMetadataModelField(featureContentType, !flagEnabled)
+                }}
+                className='ml-0'
+              />
+            </div>
+            <div className='entities-cell'>
+              <Switch
+                checked={flagRequired}
+                onChange={() => {
+                  setFlagRequired(!flagRequired)
+                  handleRequiredMetadataModelField(
+                    featureContentType,
+                    !flagRequired,
+                  )
+                }}
+                className='ml-0'
+              />
+            </div>
+          </div>
+        </div>
+      )}
+      <Button
+        disabled={!name || !typeValue}
+        onClick={() => {
+          if (isEdit) {
```

Comment:
It's really unclear what all of this is doing, can this not be simplified?

Reply:
I already simplified the logic, now it is easier to read.

Reply:
Can we pull this into a save function? It makes the markup easier to follow

Reply:
OK, done